File: packages/domain/src/sales/sale-status.ts
/** Sale lifecycle states per docs/STATE-MACHINES.md */
export enum SaleStatus {
  Pending = 'Pending',
  Confirmed = 'Confirmed',
  Cancelled = 'Cancelled',
}

export function isTerminalSaleStatus(status: SaleStatus): boolean {
  return status === SaleStatus.Confirmed || status === SaleStatus.Cancelled;
}

File: backend/crates/domain-sales/src/sale_status.rs
use std::fmt;

use serde::{Deserialize, Serialize};

/// Lifecycle state of a Sale aggregate (GLOSSARY: SaleStatus).
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub enum SaleStatus {
    Pending,
    Confirmed,
    Cancelled,
}

impl SaleStatus {
    pub fn as_str(self) -> &'static str {
        match self {
            Self::Pending => "Pending",
            Self::Confirmed => "Confirmed",
            Self::Cancelled => "Cancelled",
        }
    }
}

impl fmt::Display for SaleStatus {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{}", self.as_str())
    }
}
